packages/lib/storage/src/new/read.ts

```diff
@@ -0,0 +1,224 @@
+import {
+    getAccountManifestKey,
+    getConversationListKey,
+    getConversationManifestKey,
+    getMessageChunkKey,
+} from './keys';
+import {
+    AccountManifest,
+    Chunk,
+    ConversationList,
+    ConversationManifest,
+    Db,
+    MessageChunk,
+    ReadStrategy,
+    RemoteFetchCb,
+} from './types';
+
+/**
+ * This function retrieves a specific chunk of messages for a given conversation from the storage.
+ *
+ * @param {Db} db - The database object, which contains the account's ENS name and the sign function.
+ * @param {string} contactEnsName - The ENS name of the contact.
+ * @param {number} page - The page number of the message chunk to retrieve.
+ * @returns {Promise<MessageChunk | undefined>} A promise that resolves to the message chunk
+ *  or undefined if the chunk does not exist.
+ */
+export async function getMessageChunk(
```

Comment:
I think it would make sense to have either a class or a function that receives the db object as a constructor param instead of drilling it down every function arg

const storageReader = new StorageReader(db) / getStorageReader(db)
strageReader.read(conversation,page)